src/main/java/pl/jkap/sozzt/contract/domain/ContractFacade.java

```diff
@@ -1,55 +1,60 @@
 package pl.jkap.sozzt.contract.domain;
 
 import org.springframework.context.event.EventListener;
+import org.springframework.data.domain.PageRequest;
 import pl.jkap.sozzt.contract.dto.ContractDto;
+import pl.jkap.sozzt.contract.exception.ContractNotFoundException;
 import pl.jkap.sozzt.fileContract.event.FileUploadedSpringEvent;
 
+import java.util.List;
+import java.util.stream.Collectors;
+
 import static java.util.Objects.requireNonNull;
 
 public class ContractFacade {
```

Comment:
If you are using facade approach this should be done slightly different:
- Facade is the public interface with the contract for outside world
- Service which implements Facade has the logic part (and should be injected by profiles)
- Stub which dummy implements Facade for case where Service profile is disabled

We can jump into to quick call to discuss this, you are very close to mastering this pattern

Reply:
It would be great to discuss it :)